share/src/main/java/org/apache/kafka/server/share/persister/PersisterStateManager.java

```diff
@@ -679,6 +679,7 @@ protected void handleRequestResponse(ClientResponse response) {
                             // check retryable errors
                             case COORDINATOR_NOT_AVAILABLE:
                             case COORDINATOR_LOAD_IN_PROGRESS:
+                            case NOT_COORDINATOR:
```

Comment:
Could you please adjust behavior of `testReadStateRequestFailButCoordinatorFoundSuccessfully` to match this change?

Reply:
Done.